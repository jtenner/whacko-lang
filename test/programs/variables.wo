export fn _start(): void {
  let a = 10;
  a = 10;
}