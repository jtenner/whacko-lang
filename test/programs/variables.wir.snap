program:
  callable:[std/builtins.wo]gc.initialize;functionType:()=>voidType
(extern function)
  callable:[std/builtins.wo]gc.alloc;functionType:(u32Type,usizeType)=>ptrType
(extern function)
  callable:[std/builtins.wo]gc.store_field_ptr;functionType:(i32Type,ptrType,ptrType,ptrType)=>voidType
(extern function)
  callable:[test/programs/variables.wo]_start;functionType:()=>voidType
    block: entry~62
      64~inst = load: (variable a)
      66~inst = (i64 10)
      67~inst = store: (variable a) = (runtime i64Type 66~inst)
      return (void)

